applications/FluidDynamicsApplication/custom_utilities/cfd_utilities.h

```diff
@@ -0,0 +1,347 @@
+using ConditionType = ModelPart::ConditionType;
+using GeometryType = Geometry<NodeType>;
+
+/**
+ * @brief Calculate gauss point data for condition
+ *
+ * The method calculates and retrieves gauss point data namely
+ * gauss_weight, and gauss point shape function values at each gauss point.
+ * rNContainer matrix rows corresponds to gauss point index
+ *
+ * @param rGeometry
+ * @param rIntegrationMethod
+ * @param rGaussWeights
+ * @param rNContainer
+ */
+void CalculateConditionGeometryData(const GeometryType& rGeometry,
+                                    const GeometryData::IntegrationMethod& rIntegrationMethod,
+                                    Vector& rGaussWeights,
+                                    Matrix& rNContainer);
+
+/**
+ * @brief Calculates outwards pointing normal for a condition
+ *
+ * This method calculates outward pointing normal. The geometry should be
+ * oriented properly to have outwards pointing normal. Use
+ * TetrahedralMeshOrientationCheck before using this method to calculate normal
+ * since this process checks and corrects orientation of geometry nodes.
+ * Magnitude of normal contains area(in 3D), length (in 2D) of the condition.
+ *
+ * @tparam TDim       Dimensionality of condition. Either 2 or 3
+ * @param rNormal     Calculated normal
+ * @param rCondition  Condition
+ *
+ * @see TetrahedralMeshOrientationCheck
+ */
+
+template <unsigned int TDim>
+void CalculateConditionNormal(array_1d<double, 3>& rNormal, const ConditionType& rCondition);
+
+/**
+ * @brief Calculates wall height for condition
+ *
+ * Wall height is for given condition is calculated. Wall height is the wall normal distance
+ * between center of condition and center of parent element for that particular condition.
+ * TetrahedralMeshOrientationCheck needs to be used with ASSIGN_NEIGHBOUR_ELEMENTS_TO_CONDITIONS flag
+ * to assign parent element for each condition to use this method.
+ *
+ * @param rCondition  Condition for which wall height is calculated
+ * @param rNormal     Wall normal (outwards pointing)
+ * @return double     Wall height
+ *
+ * @see TetrahedralMeshOrientationCheck
+ */
+double CalculateConditionWallHeight(const ConditionType& rCondition,
+                                    const array_1d<double, 3>& rNormal);
+
+/**
+ * @brief Evaluates variable value at gauss point
+ *
+ * This method evaluates a variable for a given geometry at a given gauss point.
+ * The gauss point is chosen by providing proper shape function values at that
+ * gauss point via rShapeFunction.
+ *
+ * @tparam TDataType        Data type
+ * @param rGeometry         Geometry for gauss point evaluation
+ * @param rVariable         Variable
+ * @param rShapeFunction    Shape function values evaluated at gauss point
+ * @param Step              Step
+ * @return TDataType        variable value evaluated at gauss point
+ */
+template <typename TDataType>
+TDataType EvaluateInPoint(const GeometryType& rGeometry,
```

Comment:
I added the doxygen comments in the header, not in the cpp.

It would be great if we can have `EvaluateInPoint` like methods in core utility. I'm more or less recoding it everytime I want to use it :/. Where should we put it? GeometryUtilities.h?

Reply:
Here are the doxygen comments

https://github.com/KratosMultiphysics/Kratos/blob/4e0bf6d487c6568cbfc7efa96cdd79c3b69b8f59/applications/FluidDynamicsApplication/custom_utilities/cfd_utilities.h#L166-L202

Reply:
ok, sorry